src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4740,3 +4740,101 @@ void C2_MacroAssembler::udivmodL(Register rax, Register divisor, Register rdx, R
 }
 #endif
 
+void C2_MacroAssembler::float_class_check(int opcode, Register dst, XMMRegister src, Register temp) {
+
+  int32_t POS_INF = 0x7f800000;
+  int32_t KILL_SIGN_MASK = 0x7fffffff;
+
+  movdl(temp, src);
+  andl(temp, KILL_SIGN_MASK);
+  cmpl(temp, POS_INF);
```

Comment:
For IsNaN following sequence will offer better latency
     "vucomiss src_xmm, src_xmm"
     "setp r8"

Reply:
Hi @vamsi-parasa ,

I can see almost 30% improvement in you JMH micro for IsNaN through above suggested change.
Original score over AVX2 machine:

Benchmark                       Mode  Cnt  Score   Error  Units
FloatClassCheck.testIsNaN       avgt    2  0.854          ns/op
FloatClassCheck.testIsNaN:·asm  avgt         NaN            ---

With new sequence: 
Benchmark                       Mode  Cnt  Score   Error  Units
FloatClassCheck.testIsNaN       avgt    2  0.570          ns/op
FloatClassCheck.testIsNaN:·asm  avgt         NaN            ---

Reply:
Thanks Jatin! Will test it on my machine as well...